### DISCUSSION TITLE
t is not a function - when I use `@aws-sdk/client-s3` on a Cloudflare worker

### DISCUSSION THREAD

**Discussion by fayazara** [aws/aws-sdk-js-v3#6284] [Q&A]
I am pretty lost as I am not understanding where this error is coming from, I am able to get this working my S3 service working on a usual node runtime, but I just get this error `t is not a function` on a Nuxt 3 project hosted on cloudflare. 

Anybody know anything about this?

**Comment by farzadali3** [comment]
Did you try this [solution](https://github.com/aws/aws-sdk-js-v3/issues/6269#issuecomment-2232662821)?

**Reply by TheSynt4x** [reply]
I've tried it and I can confirm that on a Nuxt 3 Cloudflare Pages project, I get the same issue. On that version and even on older. Any ideas?

**Reply by bondaba** [reply]
Hello! Was a solution found for this? 

I have the same issue running Nuxt 3.13.0 with Nitro 2.9.7 with "@aws-sdk/client-s3": "^3.637.0" and  "@aws-sdk/s3-request-presigner": "^3.637.0".  

When I call getSignedUrl, I get the error "TypeError: t is not a function"

The code runs fine on Node. The issue occurs only on Cloudflare Workers.

I've also tried downgrading the version to 3.574 with no luck.

Any help/ workarounds would be greatly appreciated, thanks in advance.

**Reply by zackspear** [reply]
Another confirmation of a Nuxt project on a Cloudflare worker having the same issue. Downgrading to `3.574` does not work.

This should likely be an issue and not a discussion. Anybody know if there's an existing issue?

**Reply by fayazara** [reply]
> This should likely be an issue and not a discussion. Anybody know if there's an existing issue?

They just close the issue stating us to use AWS lambda

**Reply by codiini** [reply]
Can confirm I have the exact same issue :(

**Reply by tmorgan497** [reply]
Also getting this issue. I have a nuxt app deployed on cloudflare pages. The issue occurs when calling `getSignedUrl`.

`[nuxt] [request error] [unhandled] [500],t is not a function`

**Comment by MengLinMaker** [comment]
> I am pretty lost as I am not understanding where this error is coming from, I am able to get this working my S3 service working on a usual node runtime, but I just get this error `t is not a function` on a Nuxt 3 project hosted on cloudflare.
> 
> Anybody know anything about this?

I've experienced a similar issue in `solidstart.js`, suggesting the issue is not framework specific.
Enabling source maps will show that the problem originates from `@aws-sdk/s3-request-presigner`, specifically a `@smithy` package that imports `node:fs`

Unfortunately `node:fs` is not supported in Cloudflare workers. So calling an `node.fs` function results in the `t is not a function` error.

**Potential solution**
Use a package that doesn't use `node:fs`
I'm exploring an alternative [`aws4fetch`](https://github.com/mhart/aws4fetch). Let you know how that goes.

**Reply by MengLinMaker** [reply]
**Quick tip**
Turn on source maps when you encounter any error logs.
You'll get better error messages that tell you where the errors occurs from the source - in this case a file in `node_modules` folder.
Then you don't have to dive through the gibberish bundled code.

**Comment by MengLinMaker** [comment]
@fayazara Use [`aws4fetch`](https://github.com/mhart/aws4fetch) according to [cloudflare docs](https://developers.cloudflare.com/r2/examples/aws/aws4fetch/).

The `@aws-sdk/s3-request-presigner` package uses a `@smithy` package that requires `node:fs`, which does not exist on Cloudflare.
